### ISSUE TITLE
Generate chest contents from same seed as level layout

### ISSUE THREAD

**Issue by lheckemann** [TurningWheel/Barony#296]
Currently, the contents of chests will be regenerated when loading a saved game. This makes a particular form of scumming a lot easier (e.g. repeatedly save and reload on one of the intermediate levels that have nothing interesting besides two chests). It also makes #147 a lot harder to reproduce for debugging purposes.

**Comment by CloudyProton** [comment]
This also creates inconsistency in situations where a player explores a level and some chest items, then saves and closes the game only to later return and loadup to find that the items anticipated to be found from earlier are now completely different.

**Comment by SheridanR** [comment]
I agree in theory - not sure why chest contents aren't selected from the same pool right now. I'll have to remember to check this out.
